### PR TITLE
Add Cgroup to collect ECS instance metrics for container insights receiver

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** 

This is a split pr from this [pr](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3853)

Using Cgroup to get the cpu and mem reserved capacity through the Cgroup path.

The Cgroup path will be get based on task id, container info and cluster name.

**Testing:** <Describe what testing was performed and which tests were added.>

Unit test

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+func (ii *MockInstanceInfo) GetContainerInstanceID() string {
+	return ii.instanceID
+}
+
+func TestGetCGroupPathForTask(t *testing.T) {
+	cgroupMount := "test"
+	controller := "cpu"
+	taskID := "test1"
+	clusterName := "myCluster"
+	result, _ := getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
```
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go:57]: Don't ignore errors, even in tests.  If you expect them to be nil then assert that they are nil.

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	result, _ = getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
+	assert.Equal(t, path.Join(cgroupMount, controller, "ecs", clusterName, taskID), result)
+
+	taskID = "test5"
+	result, err := getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
+	assert.Equal(t, "", result)
+	assert.NotNil(t, err)
+}
+
+func TestGetCGroupPathFromARN(t *testing.T) {
```
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go:70]: Can you rewrite this as a table based test?  That should make it easier to read and to figure out which test case is failing.  The other tests might benefit from this as well.

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go (hunk 1 above)
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go:57]: Sure, thanks

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go (hunk 2 above)
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go:70]: Sure, will change the unit test code

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**